### PR TITLE
Fork the jemallocator crate, fix for nightly-2018-04-15

### PR DESCRIPTION
**Pull request by SimonSapin**
CC https://github.com/alexcrichton/jemallocator/pull/40, https://github.com/rust-lang/rust/pull/49669

The new version of jemallocator requires a more recent jemalloc https://github.com/alexcrichton/jemallocator/pull/34 which doesn’t build on our current Android toolchain https://github.com/jemalloc/jemalloc/issues/1175.

To avoid blocking on figuring that out, duplicate ~70 lines from jemallocator and use the older jemalloc-sys directly.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20641)

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1529,6 +1520,11 @@ dependencies = [
  "leak 0.1.2 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
+[[package]]
+name = "libc"
+version = "0.1.12"
```
Comment (Eijebong) [inline Cargo.lock]: Uh ? Could we avoid that duplication ?

File: components/allocator/lib.rs [hunk 2]
```diff
@@ -16,19 +15,90 @@ pub use platform::*;
...
+        ffi::malloc_usable_size(ptr as *const _)
     }
 
     /// Memory allocation APIs compatible with libc
     pub mod libc_compat {
-        pub use super::jemallocator::ffi::{malloc, realloc, free};
+        pub use super::ffi::{malloc, realloc, free};
+    }
+
+    pub struct Allocator;
```
Comment (emilio) [inline components/allocator/lib.rs]: Could we get an issue on file to undo this?

Comment (SimonSapin) [reply components/allocator/lib.rs]: https://github.com/servo/servo/issues/20645

**Pull request merged**